src/main/java/io/phdata/streamliner/schemadefiner/configbuilder/ScriptCommand.java

```diff
@@ -220,7 +238,111 @@ private static void build(
     }
   }
 
-  private static TableDefinition getOriginalTable(
+  private static void generateDeltaChangeSummary(
+      ConfigurationDiff configDiff,
+      Configuration config,
+      Map<String, Map<String, String>> typeMapping,
+      String outputDirectory) {
+    // this does not have deleted tables. deleted tables are already removed from this list.
+    List<TableDiff> tableDiffs = configDiff.getTableDiffs();
+    Set<SchemaChanges> validSchemaChanges = ((Jdbc) config.getSource()).getValidSchemaChanges();
+    if (validSchemaChanges == null || validSchemaChanges.isEmpty()) {
+      validSchemaChanges.addAll(Arrays.asList(SchemaChanges.values()));
+    }
+    List<TableDiff> addedTables =
+        tableDiffs.stream()
+            .filter(tableDiff -> tableDiff.existsInSource && !tableDiff.isExistsInDestination())
+            .collect(Collectors.toList());
+    List<String> addedTablesName =
+        addedTables.stream()
+            .map(tableDiff -> tableDiff.getDestinationName())
+            .collect(Collectors.toList());
+    addedOrDeletedorIncompatibleTables.put(Constants.TABLES_ADDED.value(), addedTablesName);
+    List<String> incompatibleTables =
+        tableDiffs.stream()
+            .filter(
+                tableDiff ->
+                    !tableDiff.allChangesAreCompatible(
+                        JavaHelper.convertJavaMapToScalaMap(typeMapping), validSchemaChanges))
+            .map(incompatibleTable -> incompatibleTable.getDestinationName())
+            .collect(Collectors.toList());
+    addedOrDeletedorIncompatibleTables.put(
+        Constants.TABLES_INCOMPATIBLE.value(), incompatibleTables);
+
+    // now tableDiffs will have updatedTables only.
+    tableDiffs.removeAll(addedTables);
```

Comment:
tableDiffs is a list, so this is O(n) for each entry in addedTables. These are two moderately sized lists in the worst case, so probaly won't matter.

However as a good practice if we can avoid O(n) operations easily, we should. Could you create a new LinkedHashSet and add TableDiffs list to that at the beginning of the method?

Reply:
created a Set at beginning of method.
Set<TableDiff> tableDiffs = new LinkedHashSet<>(configDiff.getTableDiffs());